### DISCUSSION TITLE
Am I using pyWhat incorrectly?

### DISCUSSION THREAD

**Discussion by fleetofthought** [bee-san/pyWhat#94] [Issues / Q&A]
Hey! I came across PyWhat on HN -- it looks super cool and I was excited to use it! I installed it on  Debian Buster system via command "pip3 install pywhat && pywhat --help" and the install went smoothly.
I'm using it as shown in the README (what /path/to/file/) but I can only get it to work on email addresses, URLs, IP addresses, and smart contract addresses. 
When I try to run it on any directories or files present on my system, no matter which sort of file I try to get pyWhat to examine, the only output I receive is "Nothing found!"

Some of the different file/path formats I've tried, unsuccessfully, to run pyWhat on:
> what this_is_bash.sh
> what heres_a_python_script.py
> what .viminfo
> what nohup.out
> what a_text_file.txt
> what '/path/to/Pycharm/projects/directory/'

In all these cases (including trying to use "pywhat" or "python3 -m PyWhat" instead of "what" to examine these files)  the only response I receive is "Nothing found!".

Am I using it incorrectly or missing something super obvious??

PyWhat seems super cool, would love to get this working.
Any advice would be much appreciated!  :)

**Comment by ghost** [comment]
Please try running pywhat on fixtures/file file (https://github.com/bee-san/pyWhat/blob/main/fixtures/file). Do you still get no output?

**Reply by fleetofthought** [reply]
Thank you - that works! I get a full response on the fixtures/file file.

Any suggestions as to what could be going on with all the other files present on my system?

**Reply by SkeletalDemise** [reply]
Do the files that you're testing actually have anything that pywhat can identify in them? Like do they have IP addresses, URLs, email addresses, etc. in them?

**Reply by fleetofthought** [reply]
Thanks! I've tried running pywhat on some other files that contain URLs and .jpg files and it works, I get a nice identification.
So I think I misunderstood some of the use cases of pywhat - I thought I'd able to run it on any file and would get at least some output as to what the file is, somewhat like the "file" command. My bad!

Thanks to @SkeletalDemise and @piatrashkakanstantinass for your clarifications, much appreciated!

**Reply by bee-san** [reply]
> file and would get at least some output as to what the file is, somewhat like the "file" command. My bad!

It actually should output that, as it can identify magic numbers. Maybe a PR broke that feature? 😟
https://github.com/bee-san/pyWhat/blob/main/pywhat/magic_numbers.py <- Related code for that

PyWhat should have full `file` support :shrug: 😞

**Reply by fleetofthought** [reply]
Hey @bee-san - First, thanks for creating this, I really do think it's neat :)

Second, thanks for pointing me to the relevant file, I'm going to take a look at it and see if I can find anything relevant. If I can get it working on my machine I'll submit a PR.

**Reply by ghost** [reply]
Pywhat prints that file is found only if there is some data that was identified. @bee-san

**Reply by bee-san** [reply]
@piatrashkakanstantinass ah :-( seems like a bug to me! it should be able to print what a file is, as that's just as important as what's in it 😄
